### DISCUSSION TITLE
Need to refresh `org-ql-view` buffers in order to see any results

### DISCUSSION THREAD

**Discussion by apc** [alphapapa/org-ql#400] [Q&A]
Very often, when I run `org-ql-search`, I get no results back when clearly there should be some. And indeed, running `org-ql-view-refresh` once shows the results I expect. Puzzlingly, this doesn't happen all the time, but I haven't yet noticed a pattern. 

I noticed this change in behavior a few months ago but I still haven't managed to find the source of it. And annoyingly, each time I've tried using `edebug` to step through `org-ql-search` and see whether I can understand what's going on, the behavior in question goes away.  

I'm now just looking for hints on places to look in order to figure out why this is happening.

**Comment by apc** [comment]
OK, just to add that the behavior in question seems to only happen when using the `:super-groups` keyword. I'm assuming this thus has something to do with when `org-super-agenda-mode` is enabled? I've tried making sure that `org-super-agenda` is loaded before running `org-ql-search` and got the same results. See e.g.: 


<img width="1347" alt="image" src="https://github.com/alphapapa/org-ql/assets/350504/f0dfbbde-96c6-490c-a87d-6b5e4e718eb5">

So I guess my question is what is the best way of configuring `org-ql` and `org-super-agenda-mode` so that `org-super-agenda-mode` is enabled when needed. At the moment, I have a `org-agenda-mode-hook` call `org-super-agenda-mode`, but maybe that's not enough?

**Comment by alphapapa** [comment]
I've never seen this problem before, and I can't think of a reason for it.  It's likely a configuration problem.  So your first step should be to try to reproduce the problem in a clean configuration.  You could use this script to help with that: https://github.com/alphapapa/makem.sh

**Reply by apc** [reply]
Thanks! I think I found the culprit, but I'm still not sure I fully understand. Right after the `*Org QL View*` buffer is displayed, the value of `org-agenda-finalize-hook` includes this function: `org-super-agenda--hide-or-show-groups`. The following minimal configuration seems to reproduce the problem I'm encountering: 

```elisp
(defvar bootstrap-version)
(let ((bootstrap-file
       (expand-file-name "straight/repos/straight.el/bootstrap.el" user-emacs-directory))
      (bootstrap-version 5))
  (unless (file-exists-p bootstrap-file)
    (with-current-buffer
        (url-retrieve-synchronously
         "https://raw.githubusercontent.com/raxod502/straight.el/develop/install.el"
         'silent 'inhibit-cookies)
      (goto-char (point-max))
      (eval-print-last-sexp)))
  (load bootstrap-file nil 'nomessage))

(setq-default use-package-enable-imenu-support t)
(setq straight-use-package-by-default t)
(setq use-package-always-ensure t)

(straight-use-package 'use-package)

(use-package org-ql)

(use-package org-super-agenda)

(add-hook 'org-agenda-finalize-hook #'org-super-agenda--hide-or-show-groups)
```

The file I'm testing with is just this: 

```
* Something
** Headline 1                                                           :tag:
:PROPERTIES:
:GROUP:    1
:END:
** Headline 2                                                           :tag:
:PROPERTIES:
:GROUP:    2
:END:
* Views
#+begin_src elisp :results output none
(org-ql-search (buffer-file-name)
  '(and (tags "tag")
        (level 2))
  :super-groups '((:auto-property "GROUP")))
#+end_src
```

Executing the block with the call to `org-ql-search` yields no results: 

<img width="581" alt="image" src="https://github.com/alphapapa/org-ql/assets/350504/f71e372c-6147-4ca1-868b-e0956a5c436c">

Now, I haven't yet figured out what is adding `org-super-agenda--hide-or-show-groups` to `org-agenda-finalize-hook` in my config. I have a bunch of spaghetti code involving `origami` and `org-super-agenda` that I'll need to sort through. But in the meantime perhaps you can tell me whether having `org-super-agenda--hide-or-show-groups` added to `org-agenda-finalize-hook` should be causing this kind of trouble? 

Thanks again for your time. And thanks for everything you share here. My workflow is so dependent on `org-super-agenda` and `org-ql` that I don't know what I'd do without them!

**Reply by alphapapa** [reply]
See https://github.com/alphapapa/org-super-agenda/blob/ee3379ae92b90c084717fb2a7614060ce12283cb/org-super-agenda.el#L365-L369

You probably enabled `org-super-agenda-hide-empty-groups`.

> Thanks again for your time. And thanks for everything you share here. My workflow is so dependent on org-super-agenda and org-ql that I don't know what I'd do without them!

Thanks for the kind words.  I'm always glad to hear that these tools are useful.  If you ever write up your workflow in a blog or something, I'd be interested to read about how you use them.

**Reply by apc** [reply]
Thanks! I now see what’s adding that function to `org-agenda-finaliza-hook`. One last question though: since there are no empty groups, why am I getting no results?

**Reply by alphapapa** [reply]
I don't know.  I don't use that feature and I didn't develop it myself.  I guess you should file a bug against `org-super-agenda`.  If you do, please include important information like software versions.  And I would recommend that you reproduce the problem using `makem.sh` instead of `straight`.
